plugins/inputs/glusterfs/glusterfs.go

```diff
@@ -0,0 +1,71 @@
+package glusterfs
+
+// glusterfs.go
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"bufio"
+	"os/exec"
+	"regexp"
+	"strconv"
+)
+
+var matchBrick = regexp.MustCompile("^Brick: (.*)$")
+var matchRead = regexp.MustCompile("Data Read: ([0-9]+) bytes$")
+var matchWrite = regexp.MustCompile("Data Written: ([0-9]+) bytes$")
+
+type GlusterFS struct {
+	Volumes []string
+}
+
+func (gfs *GlusterFS) Description() string {
+	return "Plugin reading values from the GlusterFS profiler"
+}
+
+func (gfs *GlusterFS) SampleConfig() string {
+	return "volumes = [\"volume-name\"]"
+}
+
+func (gfs *GlusterFS) Gather(acc telegraf.Accumulator) error {
+	for _, volume := range gfs.Volumes {
+		var cmdName = "sudo"
+		var cmdArgs = []string{"gluster", "volume", "profile", volume, "info", "cumulative"}
+
+		cmd := exec.Command(cmdName, cmdArgs...)
+		cmdReader, err := cmd.StdoutPipe()
+		if err != nil {
+			continue
+		}
+
+		scanner := bufio.NewScanner(cmdReader)
+		go func() {
```

Comment:
Allright thanks, just compiled a version with those changes and deployed it on our servers, we'll see how that fares. Doesn't seem to make a difference for now at least, that's a good sign.
If tomorrow the data still looks good I'll commit that.